### PR TITLE
Add java solution to clock tree

### PR DESCRIPTION
**Pull request by nathangong**

### CODE REVIEW

File: solutions/usaco-1016.mdx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+			int b = io.nextInt() - 1;
+			adj.get(a).add(b);
+			adj.get(b).add(a);
+		}
+
+		dfs(0, 0, -1);
+
+		// same mod means you can start anywhere
+		if ((sum0 % 12) == (sum1 % 12)) {
+			io.println(n);;
```
Comment (brad182) [inline solutions/usaco-1016.mdx:139]: two semicolons
```suggestion
			io.println(n);
```

File: solutions/usaco-1016.mdx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+			io.println(nodes0);
+		}
+		// no way to get all clocks pointing to 12
+		else {
+			io.println(0);
+		}
+		io.close();
+	}
+
+	static void dfs(int i, int color, int par) {
```
Comment (brad182) [inline solutions/usaco-1016.mdx:156]:
```suggestion
	static void dfs (int i, int color, int par) {
```

File: solutions/usaco-1016.mdx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+```
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+public class ClockTree {
+	static int[] clocks;
+	static List<ArrayList<Integer>> adj;
```
Comment (brad182) [inline solutions/usaco-1016.mdx:115]: probably could be an array of arraylists, but doesn't matter that much

File: solutions/usaco-1016.mdx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+	static int sum0 = 0, sum1 = 0, nodes0 = 0, nodes1 = 0;
+
+	public static void main(String[] args) throws IOException {
```
Comment (brad182) [inline solutions/usaco-1016.mdx:118]:
```suggestion
	public static void main (String[] args) throws IOException {
```

**Review by brad182 — CHANGES_REQUESTED** [review]
minor changes

File: solutions/usaco-1016.mdx (hunk 3 above)
Comment (nathangong) [reply solutions/usaco-1016.mdx:115]: Yeah I did this cause we're suposed to do polymorphism, but it also seems a bit weird so idk

File: solutions/usaco-1016.mdx (hunk 2 above)
Comment (nathangong) [reply solutions/usaco-1016.mdx:156]: I think it's convention to not leave a space in between the method name and parameters?
image

Comment (brad182) [reply solutions/usaco-1016.mdx:156]: oh

**Comment by brad182** [comment]
wait you changed it to arraylists of arraylists i was saying it was fine, just commented that I usually do array of arraylists instead

**Comment by nathangong** [comment]
Yeah I think ArrayList of ArrayLists makes more sense and is less confusing, and unfortunately Java doesn't allow making a List of Lists

**Review by brad182 — APPROVED** [review]

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
time complexity

**Comment by HarryWangATX** [comment]
> I tried to do List<List> but it didn't work, so the only other option would be List<ArrayList> but I think that may be a bit weird/confusing

wdym by doesn't work? that syntax is valid.

**Comment by nathangong** [comment]
Java throws this error when I try to do
```java List<List<Integer>> adj = new ArrayList<ArrayList<Integer>>();```
image

**Comment by nathangong** [comment]
Wait I think ```List<List<Integer>> = new ArrayList<>();``` might work

**Review by HarryWangATX — APPROVED** [review]
code looks good, havent looked at explanation yet, core needs to review it too.

**Comment by nathangong** [comment]
The explanation was approved a while back, the reason why it looks like I added everything was because I changed the file name to match the new naming standards

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**